src/hooks/useStaking.js

```diff
@@ -334,17 +334,40 @@ export function useStaking() {
     return staked
   }, [stakingContract, account])
 
-  const allowManager = useCallback(async () => {
-    if (!stakingContract || !connectedAgreementApp || !stakeManagement) {
-      return
-    }
+  const allowManager = useCallback(
+    async onDone => {
+      if (!stakingContract || !connectedAgreementApp || !stakeManagement) {
+        return
+      }
 
-    await stakingContract.allowManager(
-      connectedAgreementApp.address,
-      MAX_INT.toString(10),
-      '0x'
-    )
-  }, [connectedAgreementApp, stakingContract, stakeManagement])
+      if (onDone) {
+        const allowManagerData = encodeFunctionData(
+          stakingContract,
+          'allowManager',
+          [connectedAgreementApp.address, MAX_INT.toString(10), '0x']
+        )
+
+        const intent = [
+          {
+            data: allowManagerData,
+            from: account,
+            to: stakeManagement.stakingInstance,
+            description: 'Allow 1Hive Protocol',
+          },
+        ]
+        if (mounted()) {
+          onDone(intent)
+        }
+      } else {
```

Comment:
are we starting to avoid early returns now for some reason?

Reply:
No, but as i commented on your PR i think it's more clear for a new dev jumping in the codebase, as an early return could be overlooked in this case.

Also when it comes to performing logic conditionally i think we should always handle it in this way and only do early returns when we are returning data: e.g as we do in useMemos

Reply:
got it